source/interface/azure_iot_mqtt.h

```diff
@@ -0,0 +1,376 @@
+ *
+ * @brief Azure IoT MQTT is the MQTT interface that AzureIoT middleware depends upon.
+ *
+ * @note This interface is private and subjected to change. Currently, there is only
+ *       one implementation for this interface, which uses coreMQTT as underlying MQTT stack.
+ */
+
+#ifndef AZURE_IOT_MQTT_H
+#define AZURE_IOT_MQTT_H
+
+#include <stdint.h>
+#include <stddef.h>
+
+#include "azure_iot_mqtt_port.h"
+#include "azure_iot_transport_interface.h"
+
+#define azureiotmqttPACKET_TYPE_CONNECT        ( ( uint8_t ) 0x10U )  /** @brief CONNECT (client-to-server). */
+#define azureiotmqttPACKET_TYPE_CONNACK        ( ( uint8_t ) 0x20U )  /** @brief CONNACK (server-to-client). */
+#define azureiotmqttPACKET_TYPE_PUBLISH        ( ( uint8_t ) 0x30U )  /** @brief PUBLISH (bidirectional). */
+#define azureiotmqttPACKET_TYPE_PUBACK         ( ( uint8_t ) 0x40U )  /** @brief PUBACK (bidirectional). */
+#define azureiotmqttPACKET_TYPE_PUBREC         ( ( uint8_t ) 0x50U )  /** @brief PUBREC (bidirectional). */
+#define azureiotmqttPACKET_TYPE_PUBREL         ( ( uint8_t ) 0x62U )  /** @brief PUBREL (bidirectional). */
+#define azureiotmqttPACKET_TYPE_PUBCOMP        ( ( uint8_t ) 0x70U )  /** @brief PUBCOMP (bidirectional). */
+#define azureiotmqttPACKET_TYPE_SUBSCRIBE      ( ( uint8_t ) 0x82U )  /** @brief SUBSCRIBE (client-to-server). */
+#define azureiotmqttPACKET_TYPE_SUBACK         ( ( uint8_t ) 0x90U )  /** @brief SUBACK (server-to-client). */
+#define azureiotmqttPACKET_TYPE_UNSUBSCRIBE    ( ( uint8_t ) 0xA2U )  /** @brief UNSUBSCRIBE (client-to-server). */
+#define azureiotmqttPACKET_TYPE_UNSUBACK       ( ( uint8_t ) 0xB0U )  /** @brief UNSUBACK (server-to-client). */
+#define azureiotmqttPACKET_TYPE_PINGREQ        ( ( uint8_t ) 0xC0U )  /** @brief PINGREQ (client-to-server). */
+#define azureiotmqttPACKET_TYPE_PINGRESP       ( ( uint8_t ) 0xD0U )  /** @brief PINGRESP (server-to-client). */
+#define azureiotmqttPACKET_TYPE_DISCONNECT     ( ( uint8_t ) 0xE0U )  /** @brief DISCONNECT (client-to-server). */
+
+#define azureiotmqttNOT_CLEAN_SESSION          ( ( uint8_t ) 0x0 )    /** @brief Not MQTT clean session. */
+#define azureiotmqttCLEAN_SESSION              ( ( uint8_t ) 0x1 )    /** @brief MQTT clean session. */
+
+#define azureiotmqttNOT_RETAINED_MESSAGE       ( ( uint8_t ) 0x0 )    /** @brief Message not retained, when publishing. */
+#define azureiotmqttRETAINED_MESSAGE           ( ( uint8_t ) 0x1 )    /** @brief Message is retained, when publishing. */
+
+#define azureiotmqttNOT_DUPLICATE_MESSAGE      ( ( uint8_t ) 0x0 )    /** @brief Not Duplicate publish message. */
+#define azureiotmqttDUPLICATE_MESSAGE          ( ( uint8_t ) 0x1 )    /** @brief MQTT clean session. */
+
+typedef enum AzureIoTMQTTQoS
+{
+    eAzureIoTMQTTQoS0 = 0, /** Delivery at most once. */
+    eAzureIoTMQTTQoS1 = 1, /** Delivery at least once. */
+    eAzureIoTMQTTQoS2 = 2  /** Delivery exactly once. */
+} AzureIoTMQTTQoS_t;
+
+typedef enum AzureIoTMQTTQoS
```

Comment:
it might be a quirk of GitHub diff UX, but I see the `AzureIoTMQTTQoS` enum defined twice here - once line 46 and again 53.